pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -748,6 +753,9 @@ func applyOutlierDetection(c *cluster.Cluster, outlier *networking.OutlierDetect
 	}
 	if outlier.BaseEjectionTime != nil {
 		out.BaseEjectionTime = gogo.DurationToProtoDuration(outlier.BaseEjectionTime)
+		if outlier.BaseEjectionTime.Compare(defaultMaxEjectionTime) > 0 {
+			out.MaxEjectionTime = out.BaseEjectionTime
```

Comment:
this makes the DR doc wrong - it says that the ejection time will increase with each ejection. Do we want to cap it at 300s?

Reply:
Then it will break use intention

Reply:
Sorry, I said that wrong

What I mean is currently if they set 400s,they will actually get 400s,then 800s,then 1200s.

How, they will get 400s,400s,400s

Is that desired? Should we just set the max time to be some high number (max_int or maybe 10xbase or similar?)

Reply:
TBH, we donot rely on the ejection algorithm in production. Personally, i think this kind of agility is used by a little percentage of users. 

To have a max ejection time limit looks good to me, but i have no experience for that

Reply:
I thought the same as @howardjohn about if we should be setting max to a multiple of the base. In Envoy, I believe just seeing the base to 10m would cause it to fail (as we were seeing), and the user would need to also set the max higher to make it work. In that case, the user could make the max equal the base or something higher. We don't allow the Istio user to customize the max (if I read the api correctly), so settling on setting the max to the base is a good first pass. A followup would be to allow the Istio user to also set the max if they desired it to be greater than the base.

Reply:
I agree first this is to fix potential compatibility, whether to add a max field to the api or set some fixed value is TBD